src/unity/toolkits/one_shot_object_detection/one_shot_object_detector.hpp

```diff
@@ -0,0 +1,82 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#ifndef TURI_ONE_SHOT_OBJECT_DETECTOR_H_
+#define TURI_ONE_SHOT_OBJECT_DETECTOR_H_
+
+// TODO: Clean up imports.
+
+#include <functional>
+#include <map>
+#include <memory>
+
+#include <table_printer/table_printer.hpp>
+#include <unity/lib/extensions/ml_model.hpp>
+#include <unity/lib/gl_sframe.hpp>
+#include <unity/toolkits/coreml_export/mlmodel_wrapper.hpp>
+#include <unity/toolkits/neural_net/compute_context.hpp>
+#include <unity/toolkits/neural_net/image_augmentation.hpp>
+#include <unity/toolkits/neural_net/model_backend.hpp>
+#include <unity/toolkits/neural_net/model_spec.hpp>
+
+#include <unity/toolkits/object_detection/object_detector.hpp>
+
+namespace turi {
+namespace one_shot_object_detection {
+
+class EXPORT one_shot_object_detector: public ml_model_base {
+
+ public:
+
+  // Constructor
+  one_shot_object_detector();
+
+  // Interface exposed via Unity server
+
+  // TODO: augment -> train
+  // TODO: Add a std::map<std::string, flexible_type> options parameter to 
+  // augment when mxnet is plumbed through to C++ and we call in to the Object 
+  // Detector C++ training.
+  gl_sframe augment(gl_sframe data,
+                    std::string target_column_name,
+                    gl_sframe backgrounds,
```

Comment:
Any reason why `data` and `backgrounds` need to be SFrames instead of SArrays?

Reply:
Re: `data`, we need a column for the image and a column for the label.

`backgrounds` can be an SArray. Will resolve.

Reply:
Done.